server/controllers/shiftController.js

```diff
@@ -122,82 +122,93 @@ let ShiftController = function (
       return res.status(401).json({message: "Only admin may delete a shift"});
     }
 
-    shiftRepository
-      .getById(req.params.id, [VOLUNTEERS()])
-      .then(shift => {
-        if (!shift) {
-          res.status(400).send({message: "No such shift"});
-          return;
-        }
-        let emailClient = new EmailClient(emailClientTypes.NOREPLY);
-        let smsClient = new SMSClient();
-        shift.volunteers.forEach(volunteer => {
-          let message = constructCancelShiftMessage(volunteer, shift);
-          if (volunteer.user.contactPreferences.email) {
-            emailClient.send(volunteer.user.email, "Shift cancelled", message);
-          }
-          if (volunteer.user.contactPreferences.text) {
-            smsClient.send(volunteer.user.telephone, message);
-          }
-        });
-        return shiftRepository.removeById(req.params.id);
-      })
-      .then(shift =>
-        res.status(200).json({message: "Successfully deleted", shift: shift})
-      )
+    // Check shift exists
+    let shift;
+    try {
+      shift = await shiftRepository.getById(req.params.id, [VOLUNTEERS()]);
+    } catch (err) {
+      return res.status(500).json({message: errorMessage(err)});
+    }
+    if (!shift) {
+      return res.status(400).send({message: "No such shift"});
+    }
+    // Notify volunteers
+    let emailClient = new EmailClient(emailClientTypes.NOREPLY);
```

Comment:
nit: `emailClient`, `smsClient`, and `message` should all be `const`.

Reply:
Fixed in next commit